sandy/core/cov.py

```diff
@@ -470,14 +496,27 @@ def invert(self):
         0 0.00000e+00 0.00000e+00 0.00000e+00
         1 0.00000e+00 5.00000e-01 0.00000e+00
         2 0.00000e+00 0.00000e+00 3.33333e-01
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> np.round(S.invert(sparse=True).data,2)
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 5.00000e-01 0.00000e+00
+        2 0.00000e+00 0.00000e+00 3.30000e-01
         """
         index, columns = self.data.index, self.data.columns
-        M_nonzero_idxs, M_reduce = self._reduce_size()
-        unit = np.identity(len(M_reduce))
-        M_reduce_inv = splu(csc_matrix(M_reduce)).solve(unit)
-        data = CategoryCov._restore_size(M_nonzero_idxs, M_reduce_inv,
-                                         len(self.data))
-        M_inv = pd.DataFrame(data.data,
+        if sparse is True:
+            cov = self.to_sparse(method='coo_matrix').tocsc()
+            lu = spsl.splu(cov)
+            eye = np.eye(cov.shape[0])
+            data = lu.solve(eye)
+        else:
+            M_nonzero_idxs, M_reduce = self._reduce_size()
+            unit = np.identity(len(M_reduce))
+            M_reduce_inv = splu(csc_matrix(M_reduce)).solve(unit)

```

Comment:
This is also using a sparse matrix, although we are in `else`

Reply:
I thought that with sparse matrix is not need to reduce and restore size.

Reply:
I mean, we call csc_matrix that is sparse, although this is the solution for the non-sparse matrix.
Should we just do `splu(M_reduce).solve(unit)`?

Reply:
@AitorBengoechea , i still don't get why calling csc_matrix

Reply:
https://stackoverflow.com/questions/21693613/is-it-possible-to-compute-an-inverse-of-sparse-matrix-in-python-as-fast-as-in-ma 
I found here in the second answer. If you put csr, you will recibe the following SparseEfficiencyWarning: splu requires CSC matrix format